bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,469 @@
+
+```
+<cloud>
+  <enable-jgroups-password>true</enable-jgroups-password>
+</cloud>
+```
+Configure for kubernetes execution:
+
+```
+<cloud>
+  <type>kubernetes</type>
+</cloud>
+```
+
+===== WildFly OpenShift operator
+
+The WildFly OpenShift operator can be used to manage deployments based on image containing a WildFly bootable JAR.
+At boot time, the WildFly bootable JAR dumps in the file ```/opt/jboss/container/wildfly-bootable-jar/install-dir``` its installation path.
+This information is required by the WildFly OpenShift operator to retrieve transaction logs and call into WildFly CLI.
+
+===== JKube Maven plugin
+
+The JKube Maven Plugin (version 1.0.0-rc-1 for now) has been evolved with a generator that recognizes the bootable JAR maven plugin.
+An https://github.com/eclipse/jkube/tree/v1.0.0-rc-1/quickstarts/maven/wildfly-jar[example] of Bootable JAR Maven plugin and JKube Maven Plugin to deploy
+application on OpenShift and Kubernetes. More information on the JKube Maven plugins can be found in JKube https://www.eclipse.org/jkube/docs/[documentation] 
+
+=== Nice-to-Have Requirements
+
+* Add the ability to copy content in the server during build. That is a common requirement (e.g.: auth properties files).
+* Ability to generate a runtime Maven repository in order to resolve the server artifacts from Maven local cache. 
+Huge benefit in term of JAR size and boot time (4/5 time faster, around 160ms to start vs 700 ms). 
+Specifically in a docker/openshift context, relying on slim server + Maven repository speeds up startup without impacting image size.
+Eg: java -Dmaven.repo.local=/maven-repo -jar myapp-bootable.jar
+
+
+=== Non-Requirements
+
+* Offer a new way to configure the server (e.g.: Thorntail yaml file).
+* Package a custom standalone XML file. Standalone XML file is generated during build by the Maven plugin and can't be replaced. 
+* Usage of a different log manager than JBoss log manager is out of scope.
+* No domain support.
+* Auto-detection of Galleon layers based on user application is out of scope.
+
+== Implementation Plan
+
+* Evolve wildfly-core with a runtime to boot the bootable JAR.
+* Develop new Maven plugin.
+
+== Test Plan
+
+* Maven plugin https://github.com/wildfly-extras/wildfly-jar-maven-plugin/tree/master/tests[tests] (in plugin repo)
+* Bootable runtime https://github.com/wildfly/wildfly-core/tree/master/bootable-jar/runtime/src/test/java/org/wildfly/core/jar/runtime[tests] (in wildfly-core repo)
+* Wildfly-core https://github.com/wildfly/wildfly-core/blob/master/testsuite/pom.xml#L425[tests] (in wildfly-core repo). 
+Run existing tests (when applicable) against bootable JAR using the _-Dts.bootable_ maven profile: ```cd <wildfly repo>/testsuite; mvn clean install -Dts.bootable```
+
+== Community Documentation
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/download/2.0.0.Beta1/index.html[Maven plugin documentation]
```

Comment:
I think it probably should be recorded somewhere as a requirement that examples and documentation should be using Elytron security where required and not the legacy security subsystem or legacy security realms as both of these are deprecated for removal in a future release.

Reply:
Thank-you. Done.